ml-agents/mlagents/trainers/optimizer/tf_optimizer.py

```diff
@@ -140,8 +140,9 @@ def create_reward_signals(
                 self.reward_signals[reward_signal.value].update_dict
             )
 
+    @classmethod
     def create_optimizer_op(
-        self, learning_rate: tf.Tensor, name: str = "Adam"
+        cls, learning_rate: tf.Tensor, name: str = "Adam"
```

Comment:
This didn't need the actual instance, but can still be overridden by derived classes.

Reply:
👍